resources/models/reservation.py

```diff
@@ -290,10 +290,24 @@ def can_modify(self, user):
             return self.resource.can_modify_paid_reservations(user)
 
         # reservations that need manual confirmation and are confirmed cannot be
-        # modified or cancelled without reservation approve permission
+        # modified or cancelled without reservation approve permission. However, if
+        # resource allows cancelling and the if cancelled before the allowed number of
+        # days, it's possible to cancel the reservation for resource that are free i.e.
+        # The ones that can be used without paying.
+        cancellation_min_days_in_advance = self.resource.cancellation_min_days_in_advance
+        cancellation_date_in_range = (self.begin - datetime.datetime.now(timezone.utc)).days >= cancellation_min_days_in_advance
         cannot_approve = not self.resource.can_approve_reservations(user)
```

Comment:
Sorry if this was unclear but I meant all of these new lines would be moved 301, as otherwise they will just do unnecessary calculations.